### PR TITLE
Draft: Disable extension

### PR DESCRIPTION
**Pull request by Laurens256**
Hi, I created the functionality to disable extensions without removing them from the list. It works by adding a `is_enabled` key to the extension preferences, which defaults to true if not found.

There were some things I couldn't figure out on my own. in `_migrate_user_prefs` I added the `is_enabled` key, but when migrating with this code, preferences and triggers become empty objects. After manually setting these back, everything does work as expected again, just the migration itself is broken. I couldn't for the life of me figure out why (i tried a lot of stuff to fix this)

Also when running the tests I get the following error which I couldn't find where it was coming from
```
would reformat .../Ulauncher/ulauncher/modes/extensions/ExtensionManifest.py

Oh no! 💥 💔 💥
1 file would be reformatted, 147 files would be left unchanged.
```

Here's a video showing how it works
cinnamon-2023-10-27T104020+0200.webm

Resolves https://github.com/Ulauncher/Ulauncher/issues/677

### CODE REVIEW

File: ulauncher/modes/extensions/ExtensionManifest.py [hunk 1]
```diff
@@ -188,8 +189,17 @@ def apply_user_preferences(self, user_prefs: dict):
...
+        return self.is_enabled
+
+    def toggle_is_enabled(self, enable = None):
+        self.is_enabled = enable or not self.is_enabled
+
     @classmethod
     def load_from_extension_id(cls, ext_id: str):
         manifest = super().load(f"{PATHS.EXTENSIONS}/{ext_id}/manifest.json")
         manifest.apply_user_preferences(JsonConf.load(f"{PATHS.EXTENSIONS_CONFIG}/{ext_id}.json"))
+        print(f"{PATHS.EXTENSIONS_CONFIG}/{ext_id}.json")
```
Comment (Laurens256) [inline ulauncher/modes/extensions/ExtensionManifest.py:204]: Left this in on accident

**Comment by friday** [comment]
would reformat ... is https://github.com/psf/black and you can autoformat with that fast and easy.. I will check the rest tonight.

**Review by friday — COMMENTED** [review]
ExtensionManifest is for the manifest only. Adding it there basically means that extension devs can specify whether their extensions will be enabled by default when they are installed.

So I reverted and rewrote the Python parts, because it was easier to just do it than trying to guide you though how to.

About the rest I think I would prefer it to change the toggle method to a setter, meaning you pass the boolean. That way the state can never get out of sync.

Also, I plan to rewrite the whole preferences window #1257 and also want to change it so extensions only run on demand #1063, so this feature isn't a prio for me that I would have introduced at this point, since we're gonna have to redo a lot/most of it. But it's also not a huge problem.

**Comment by Laurens256** [comment]
haha sorry, I knew I didn't understand a lot of what was going on :) just wanted to see if I could get it working. Feel free to use / not use whatever you want obviously, hope I didn't cause you too much of a headache ;)

**Comment by friday** [comment]
> haha sorry, I knew I didn't understand a lot of what was going on :) just wanted to see if I could get it working. Feel free to use / not use whatever you want obviously, hope I didn't cause you too much of a headache ;)

No, your contributions are very much appreciated. We don't get a lot of contributions, and if we do it's mostly smaller ones, trying to change a line in some isolated place. I can tell that you are making an effort to keep our standards. Some things are just too hard to understand without having the whole history though.

If you want to contribute more you can always ask us if we want the change, if it's a good time for it and if you need help with finding where to change something :)

I will merge this now. I pushed another change. It says it's failing, but that's actually the base branch.

**Pull request merged**